app/javascript/mastodon/components/column_header.tsx

```diff
@@ -0,0 +1,309 @@
+import PropTypes from 'prop-types';
+import { PureComponent, useCallback } from 'react';
+
+import { FormattedMessage, injectIntl, defineMessages } from 'react-intl';
+import type { IntlShape } from 'react-intl';
+
+import classNames from 'classnames';
+import type { RouteComponentProps } from 'react-router-dom';
+import { withRouter } from 'react-router-dom';
+
+import { ReactComponent as AddIcon } from '@material-symbols/svg-600/outlined/add.svg';
+import { ReactComponent as ArrowBackIcon } from '@material-symbols/svg-600/outlined/arrow_back.svg';
+import { ReactComponent as ChevronLeftIcon } from '@material-symbols/svg-600/outlined/chevron_left.svg';
+import { ReactComponent as ChevronRightIcon } from '@material-symbols/svg-600/outlined/chevron_right.svg';
+import { ReactComponent as CloseIcon } from '@material-symbols/svg-600/outlined/close.svg';
+import { ReactComponent as TuneIcon } from '@material-symbols/svg-600/outlined/tune.svg';
+
+import type { IconProp } from 'mastodon/components/icon';
+import { Icon } from 'mastodon/components/icon';
+import {
+  ButtonInTabsBar,
+  useColumnsContext,
+} from 'mastodon/features/ui/util/columns_context';
+import type { ContextWithIdentity } from 'mastodon/utils/identity';
+
+import { useAppHistory } from './router';
+
+const messages = defineMessages({
+  show: { id: 'column_header.show_settings', defaultMessage: 'Show settings' },
+  hide: { id: 'column_header.hide_settings', defaultMessage: 'Hide settings' },
+  moveLeft: {
+    id: 'column_header.moveLeft_settings',
+    defaultMessage: 'Move column to the left',
+  },
+  moveRight: {
+    id: 'column_header.moveRight_settings',
+    defaultMessage: 'Move column to the right',
+  },
+});
+
+interface BackButtonProps {
+  pinned?: boolean;
+  show?: boolean;
+}
+
+const BackButton = ({ pinned, show }: BackButtonProps) => {
+  const history = useAppHistory();
+  const { multiColumn } = useColumnsContext();
+
+  const handleBackClick = useCallback(() => {
+    if (history.location.state?.fromMastodon) {
+      history.goBack();
+    } else {
+      history.push('/');
+    }
+  }, [history]);
+
+  const showButton =
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    history &&
+    !pinned &&
+    ((multiColumn && history.location.state?.fromMastodon) || show);
+
+  if (!showButton) return null;
+
+  return (
+    <button onClick={handleBackClick} className='column-header__back-button'>
+      <Icon
+        id='chevron-left'
+        icon={ArrowBackIcon}
+        className='column-back-button__icon'
+      />
+      <FormattedMessage id='column_back_button.label' defaultMessage='Back' />
+    </button>
+  );
+};
+
+interface ColumnHeaderProps extends RouteComponentProps {
```

Comment:
Why are you using `RouterComponentProps` here? `<ColumnHeader>` should never be called as `<Route component=ColumnHeader>`

Reply:
Since `ColumnHeaderInternal` is passed to `injectIntl(withRouter(ColumnHeaderInternal))`, 

```ts
export function withRouter<P extends RouteComponentProps<any>, C extends React.ComponentType<P>>(
```

...so the props really do extend from `RouterComponentProps`.